delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/ResourceRepresentation.scala

```diff
@@ -0,0 +1,99 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model
+
+import cats.Order
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.decoder.JsonLdDecoder
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.decoder.JsonLdDecoderError.ParsingFailure
+import io.circe.{Decoder, Encoder}
+
+/**
+  * Enumeration of representations for resources.
+  */
+sealed trait ResourceRepresentation extends Product with Serializable {
+
+  /**
+    * Default extension for the format
+    */
+  def extension: String
+
+}
+
+object ResourceRepresentation {
+
+  /**
+    * Source representation of a resource.
+    */
+  final case object SourceJson extends ResourceRepresentation {
+    override def extension: String = ".json"
+
+    override val toString: String = "source"
+  }
+
+  /**
+    * Compacted JsonLD representation of a resource.
+    */
+  final case object CompactedJsonLd extends ResourceRepresentation {
+    override def extension: String = ".json"
+
+    override val toString: String = "compacted"
+  }
+
+  /**
+    * Expanded JsonLD representation of a resource.
+    */
+  final case object ExpandedJsonLd extends ResourceRepresentation {
+    override def extension: String = ".json"
+
+    override val toString: String = "expanded"
+  }
+
+  /**
+    * NTriples representation of a resource.
+    */
+  final case object NTriples extends ResourceRepresentation {
+    override def extension: String = ".nt"
+
+    override val toString: String = "n-triples"
+  }
+
+  final case object NQuads extends ResourceRepresentation {
+    override def extension: String = ".nq"
+
+    override val toString: String = "n-quads"
+  }
+
+  /**
+    * Dot representation of a resource.
+    */
+  final case object Dot extends ResourceRepresentation {
+    override def extension: String = ".dot"
+
+    override val toString: String = "dot"
+  }
+
+  private def parse(value: String) =
+    value match {
+      case SourceJson.toString      => Right(SourceJson)
+      case CompactedJsonLd.toString => Right(CompactedJsonLd)
+      case ExpandedJsonLd.toString  => Right(ExpandedJsonLd)
+      case NTriples.toString        => Right(NTriples)
+      case NQuads.toString          => Right(NQuads)
+      case Dot.toString             => Right(Dot)
+      case other                    => Left(s"$other is not a valid representation")
+    }
```

Comment:
String literals are more appropriate here, as we probably don't want the API to change if we rename one of these (unless we're sure these are covered by tests)

Reply:
The `toString` method is overriden.